atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceTest.java

```diff
@@ -0,0 +1,435 @@
+import org.junit.runners.Parameterized;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(Parameterized.class)
+public final class TrackingKeyValueServiceTest {
+    private static final long TIMESTAMP = 14L;
+    private static final int MAX_RESULTS = 25;
+    private static final ImmutableMap<Cell, Long> TIMESTAMP_BY_CELL_MAP =
+            TrackingKeyValueServiceTestUtils.createLongByCellMapWithSize(121, TIMESTAMP);
+    private static final Iterable<byte[]> ROWS =
+            Stream.generate(() -> new byte[100]).limit(3).collect(Collectors.toUnmodifiableList());
+    private static final TableReference TABLE_REF = TableReference.createWithEmptyNamespace("table1.namespace1");
+    private static final RangeRequest RANGE_REQUEST = RangeRequest.all();
+    private static final ImmutableList<RangeRequest> RANGE_REQUESTS = ImmutableList.of(
+            RangeRequest.all(),
+            TrackingKeyValueServiceTestUtils.createRangeRequest((byte) 1),
+            TrackingKeyValueServiceTestUtils.createRangeRequest((byte) 2));
+    private static final int CELL_BATCH_HINT = 2;
+    private static final byte[] START_BYTES = TrackingKeyValueServiceTestUtils.createBytesWithSize(5, (byte) 1);
+    private static final byte[] END_BYTES = TrackingKeyValueServiceTestUtils.createBytesWithSize(10, (byte) 1);
+    private static final ColumnSelection COLUMN_SELECTION = ColumnSelection.create(ImmutableSet.of(
+            TrackingKeyValueServiceTestUtils.createBytesWithSize(100, (byte) 0),
+            TrackingKeyValueServiceTestUtils.createBytesWithSize(101, (byte) 1),
+            TrackingKeyValueServiceTestUtils.createBytesWithSize(102, (byte) 2)));
+    private static final ColumnRangeSelection COLUMN_RANGE_SELECTION = new ColumnRangeSelection(START_BYTES, END_BYTES);
+    private static final BatchColumnRangeSelection BATCH_COLUMN_RANGE_SELECTION =
+            BatchColumnRangeSelection.create(COLUMN_RANGE_SELECTION, CELL_BATCH_HINT);
+    private static final CandidateCellForSweepingRequest CANDIDATE_CELL_FOR_SWEEPING_REQUEST =
+            ImmutableCandidateCellForSweepingRequest.builder()
+                    .startRowInclusive()
+                    .maxTimestampExclusive(TIMESTAMP)
+                    .shouldCheckIfLatestValueIsEmpty(false)
+                    .shouldDeleteGarbageCollectionSentinels(false)
+                    .build();
+    private static final Set<Cell> CELLS = ImmutableSet.of(
+            TrackingKeyValueServiceTestUtils.createCellWithSize(10, (byte) 0),
+            TrackingKeyValueServiceTestUtils.createCellWithSize(20, (byte) 1),
+            TrackingKeyValueServiceTestUtils.createCellWithSize(30, (byte) 2));
+
+    @Parameterized.Parameters(name = "size = {0}")
+    public static Object[] size() {
+        // divisible by 12 to allow creation of TableReference of wanted sizes
+        return new Object[] {12 * 129, 12 * 365, 12 * 411};
```

Comment:
I cannot use bigger numbers as I generate most lists to have 3 elements and this runs into limits with bigger sizes (i.e. cell name cannot have more than 1500 chars)

The way around this is to have more than 3 elements depending on the size but I think that's too much logic for a test (this test has enough complexity as is IMO)